Framework/PythonInterface/test/python/plugins/algorithms/FindGlobalBMatrixTest.py

```diff
@@ -0,0 +1,150 @@
+import unittest
+from testhelpers import create_algorithm
+from mantid.simpleapi import (FindGlobalBMatrix, AnalysisDataService, SetGoniometer, SetUB, CreatePeaksWorkspace,
+                              LoadEmptyInstrument, CloneWorkspace, TransformHKL)
+import numpy as np
+from FindGoniometerFromUB import getR
+
+
+def add_peaksHKL(ws_list, Hs, Ks, L):
+    for h in Hs:
+        for k in Ks:
+            for peaks in ws_list:
+                pk = peaks.createPeakHKL([h, k, L])
+                peaks.addPeak(pk)
+
+
+def getBMatrix(ws):
+    return ws.sample().getOrientedLattice().getB()
+
+
+def getUMatrix(ws):
+    return ws.sample().getOrientedLattice().getU()
+
+
+class FindGlobalBMatrixTest(unittest.TestCase):
+
+    def setUp(self):
+        # laod empty instrument so can create a peak table
+        self.ws = LoadEmptyInstrument(InstrumentName='SXD', OutputWorkspace='empty_SXD')
+        axis = self.ws.getAxis(0)
+        axis.setUnit("TOF")
+
+    def tearDown(self):
+        AnalysisDataService.clear()
+
+    def test_finds_average_lattice_parameter(self):
+        # create two peak tables with UB corresponding to different lattice constant, a
+        peaks1 = CreatePeaksWorkspace(InstrumentWorkspace=self.ws, NumberOfPeaks=0, OutputWorkspace="SXD_peaks1")
+        UB = np.diag([1.0/3.8, 0.25, 0.1])  # alatt = [3.8, 4, 10]
+        SetUB(peaks1, UB=UB)
+        peaks2 = CreatePeaksWorkspace(InstrumentWorkspace=self.ws, NumberOfPeaks=0, OutputWorkspace="SXD_peaks2")
+        UB = np.diag([1.0/4.2, 0.25, 0.1])  # alatt = [4.2, 4, 10]
+        SetUB(peaks2, UB=UB)
+        # Add some peaks
+        add_peaksHKL([peaks1, peaks2], range(0, 3), range(0, 3), 4)
+
+        FindGlobalBMatrix(PeakWorkspaces=[peaks1, peaks2], a=4.1, b=4.2, c=10, alpha=88, beta=88, gamma=89,
+                          Tolerance=0.15)
+
+        # check lattice  - should have average a=4.0
+        self.assert_lattice([peaks1, peaks2], 4.0, 4.0, 10.0, 90.0, 90.0, 90.0, delta_latt=2e-2, delta_angle=2.5e-1)
+        self.assert_matrix([peaks1], getBMatrix(peaks2), getBMatrix, delta=1e-10)  # should have same B matrix
+        self.assert_matrix([peaks1, peaks2], np.eye(3), getUMatrix, delta=5e-2)
+
+    def test_handles_inaccurate_goniometer(self):
+        # create two peak tables with UB corresponding to different lattice constant, a
```

Comment:
On a different branch but need to remove this comment- it applies to  the test above